### ISSUE TITLE
Improve navigation order for Screenreaders when using Slivers to serve Material Guidelines

### ISSUE THREAD

**Issue by CXgbw** [flutter/flutter#178154]
### Use case

[According to Material Design, the **focus order of elements** in a Scaffold should be AppBar -> Navigation Bar -> FAB -> Content](https://m3.material.io/components/floating-action-button/accessibility#c9dfe413-7661-43f5-988b-bce6da2bc47e)

This is easy to do as long **as your AppBar is in the Scaffold's appbar slot as now appBar, body and fab are siblings**. You can just set the semantics sortKey and the screenreader will navigate accordingly.

However, when it comes to slivers, the AppBar is part of the body's CustomScrollView.
You will have to scroll through the entire body to get to the fab. That's very bad user experience. Maybe I am missing something, because this seems to be quite a fundamental problem with how sliver pages are built.

There shouldn't be the need for some hacky solution to this.

```Scaffold(
      body: CustomScrollView(
        slivers: [
          SliverAppBar(
            title: const Text('My Title'),
          ),
          SliverList(
            delegate: SliverChildBuilderDelegate(
                  (context, index) => ListTile(
                    title: Text('Entry $index'),
                  ),
              childCount: 20,
            ),
          ),
        ],
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {},
        tooltip: "Add To List",
        child: const Icon(Icons.add),
      ),
    );
```

### Proposal

The best solution would be to be able to set the order across the entire screen and not just siblings.

**Comment by CXgbw** [comment]
> [@CXgbw](https://github.com/CXgbw) Does https://api.flutter.dev/flutter/semantics/OrdinalSortKey-class.html help your case further?

This works without slivers in the basic scaffold layout.
```@override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Semantics(
          sortKey: OrdinalSortKey(0.0),
          child: const Text('My Title'),
        ),
      ),
      body: Semantics(
        sortKey: OrdinalSortKey(2.0),
        child: ListView(
          children: List.generate(
            20,
            (index) => ListTile(
              title: Text('Entry $index'),
            ),
          ),
        ),
      ),
      floatingActionButton: Semantics(
        sortKey: OrdinalSortKey(1.0),
        child: FloatingActionButton(
          onPressed: () {},
          child: Icon(Icons.add),
        ),
      ),
      bottomNavigationBar: BottomNavigationBar(items: [
        BottomNavigationBarItem(icon: Icon(Icons.home), label: "Home"),
        BottomNavigationBarItem(icon: Icon(Icons.settings), label: "Settings"),
      ]),
    );
```

But doesn't with CustomScrollView because appbar, body and fab are not siblings as far as I know.

```Scaffold(
      body: CustomScrollView(slivers: [
        SliverAppBar(
          title: Semantics(sortKey: OrdinalSortKey(0.0), child: const Text('My Title')),
        ),
        SliverList(
          delegate: SliverChildBuilderDelegate(
            (context, index) => Semantics(
              sortKey: OrdinalSortKey(2.0 + index.toDouble()),
              child: ListTile(
                title: Text('Entry $index'),
              ),
            ),
            childCount: 20,
          ),
        ),
      ]),
      floatingActionButton: Semantics(
        sortKey: OrdinalSortKey(1.0),
        child: FloatingActionButton(
          onPressed: () {},
          child: Icon(Icons.add),
        ),
      ),
      bottomNavigationBar: BottomNavigationBar(items: [
        BottomNavigationBarItem(icon: Icon(Icons.home), label: "Home"),
        BottomNavigationBarItem(icon: Icon(Icons.settings), label: "Settings"),
      ]),
    )
```

**Comment by chunhtai** [comment]
I think the guideline is talking about keyboard focus not the a11y focus. a11y focus should follow screen order in most cases.
